modApiServer/src/org/aion/api/server/types/Blk.java

```diff
@@ -94,6 +96,36 @@ public static Object AionBlockToJson(
         return obj;
     }
 
+    public static JSONObject aionBlockDetailsToJson(Block genericBlock,
+        List<AionTxInfo> aionTxInfoList, Long previousTimestamp, BigInteger totalDifficulty, BigInteger blockReward){
+
+        JSONObject obj = AionBlockOnlyToJson(genericBlock, totalDifficulty);
+
+        if(obj == null){
+            return null;
+        }
+
+        long blockTime;
+        if (previousTimestamp == null){
+            blockTime = 0;
```

Comment:
would null make more sense than 0 here?

Reply:
I updated the implementation to return 0 if the requested block was the genesis block and otherwise return null